src/trace-agent.js

```diff
@@ -66,7 +66,8 @@ function TraceAgent(config, logger) {
 TraceAgent.prototype.stop = function() {
   hooks.deactivate();
   pluginLoader.deactivate();
-  cls.destroyNamespace();
+  this.config.enabled = false;
+  this.policy = tracingPolicy.createTracePolicy(this.config);
```

Comment:
For plugins that don't unpatch anything, this would let them at the very least short-circuit (since spans will never be created).